src/android/interoperability/cpp/type-mapping.md

```diff
@@ -0,0 +1,26 @@
+# Additional Types
+
+| Rust Type         | C++ Type             |
+|-------------------|----------------------|
+| `String`          | `rust::String`       |
+| `&str`            | `rust::Str`          |
+| `CxxString`       | `std::string`        |
+| `&[T]`/`&mut [T]` | `rust::Slice`        |
+| `Box<T>`          | `rust::Box<T>`       |
+| `UniquePtr<T>`    | `std::unique_ptr<T>` |
+| `Vec<T>`          | `rust::Vec<T>`       |
+| `CxxVector<T>`    | `std::vector<T>`     |
+
+<details>
+
+* These types can be used in the fields of shared structs and the arguments and
+  returns of extern functions.
+* Note that Rust's `String` does not map directly to `std::string`. There are a
+  few reasons for this:
+  * `std::string` does not uphold the utf-8 invariant that `String` requires.
```

Comment:
```suggestion
  * `std::string` does not uphold the UTF-8 invariant that `String` requires.
```

Reply:
Thanks for including the explanation here! I think it's important so people get a nice mental model of this.